### PR TITLE
set num_workers to 1 for esm2 tests

### PR DESCRIPTION
**Pull request by pstjohn**
Should help to address errors like https://prod.blsm.nvidia.com/bionemo-external-bionemo-fw/job/test_pytest/838/console 
```14:14:12  E       RuntimeError: DataLoader worker (pid 8445) is killed by signal: Aborted.```
when we use too many resources on the CI machine

### CODE REVIEW

File: sub-packages/bionemo-esm2/tests/bionemo/esm2/model/finetune/test_finetune.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ def pretrain_data_module(dummy_protein_dataset, dummy_parquet_train_val_inputs):
         micro_batch_size=4,
         min_seq_length=None,
         max_seq_length=1024,
+        num_workers=1,
```
Comment (jstjohn) [inline sub-packages/bionemo-esm2/tests/bionemo/esm2/model/finetune/test_finetune.py:57]: Why not 0? If you get a pin error you could add a `pin_memory=num_workers>0` in your datamodule kind of thing.

Comment (pstjohn) [reply sub-packages/bionemo-esm2/tests/bionemo/esm2/model/finetune/test_finetune.py:57]: It interleaves data loading and computation, and might catch multiprocessing bugs? But I'm not picky either way

Comment (jstjohn) [reply sub-packages/bionemo-esm2/tests/bionemo/esm2/model/finetune/test_finetune.py:57]: the benefit to 0 in testing would be that you can run pdb and land in a data thread if that's where an error happens. I guess you're right that it has that drawback on not testing a standard case as well though.

Comment (jstjohn) [reply sub-packages/bionemo-esm2/tests/bionemo/esm2/model/finetune/test_finetune.py:57]: whatever you want to do is fine here.

**Review by jstjohn — APPROVED** [review]

File: sub-packages/bionemo-esm2/tests/bionemo/esm2/model/finetune/test_finetune.py (hunk 1 above)
Comment (pstjohn) [reply sub-packages/bionemo-esm2/tests/bionemo/esm2/model/finetune/test_finetune.py:57]: yeah let's keep it at 1 for now -- if we do get a reproducible error here, we can always set it to zero for debugging.

**Review by polinabinder1 — APPROVED** [review]

**Pull request merged**